fixed creator tag entering inside card on smaller devices and also fi…

Morai-C opened the pull request:
…xed horizontal scroll and overflow

## Summary

Description of PR here...
Closes #364, Closes #365

## Changes

adjusted styling for card to fit creator tag on small screen devices
included breakpoints in styling to fix problem of horizontal scroll and overflow on smaller devices

## Screenshots


![Screenshot (112)](https://user-images.githubusercontent.com/77807331/161173919-dac5d7ff-1d2d-4116-92d8-8cb632669192.png)
![Screenshot (113)](https://user-images.githubusercontent.com/77807331/161173932-4c4b14f7-9899-453d-bbd0-4b004153ee08.png)

 @tuxology please this is ready for review please check and also merge

Morai-C committed:
fixed creator tag entering inside card on smaller devices and also fixed horizontal scroll and overflow

tuxology commented on zubhub_frontend/zubhub/package.json:
```diff
@@ -23,10 +23,10 @@
     "react-dom": "^17.0.1",
     "react-i18next": "^11.8.5",
-    "react-quill": "^1.3.5",
+    "react-quill": "^0.0.2",
```
Any reason, why we downgraded this package?

Morai-C replied:
in order to access auto update on save i used npm to start the server on my terminal it was downgraded when i ran npm install command because it wasn't compatible with my version of npm. but anyother person who uses npm start on the frontend it's going to install the packages compatible with their version.

NdibeRaymond replied:
@Morai-C , perhaps you should upgrade your npm version?

tuxology commented on zubhub_frontend/zubhub/src/assets/js/styles/components/project/projectStyles.js:
```diff
@@ -1,6 +1,12 @@
 const styles = theme => ({
   root: {
-    maxWidth: 345,
+    [theme.breakpoints.down('md')]: {
```
What does this do? Maybe I don't know. Just trying to learn. @NdibeRaymond probably can give us some insight

Morai-C replied:
@tuxology  okay well Material UI also has media query breakpoints just like bootstrap so --> .down('md') <-- means on screens from medium sizes to extra small it should apply that style to them. basically the margins were causing the horizontal overflow on medium devices downwards so i removed them on those devices with the them.breakpoints styling..

kamthamc replied:
@tuxology  Any screen size less than [900px](https://mui.com/customization/breakpoints/) will have 0 marginLeft and marginRight.

@Morai-C You don't need to specify units if the value is zero. It can be simply `marginLeft: 0`.

If you are setting margin left and right values you could also use `margin-inline`
https://developer.mozilla.org/en-US/docs/Web/CSS/margin-inline

Morai-C replied:
@kamthamc thanks for this.. i'm guessing the margin-inline would work the same way marginHorizontal on react native works?

kamthamc replied:
It should work in the same way. `margin-inline` is a standard css property. And I think `marginHorizontal` is specific to react-native

tuxology reviewed (commented):
Need someone else to look at it also

NdibeRaymond commented on zubhub_frontend/zubhub/package.json:
```diff
@@ -23,10 +23,10 @@
     "react-redux": "^7.2.2",
     "react-router-dom": "^5.2.0",
-    "react-scripts": "4.0.3",
+    "react-scripts": "^5.0.0",
```
it also looks like you upgraded the react script version here. This is not related to the changes you are working on. Like you said, if this was imposed on you by your npm installation, perhaps you should upgrade (or downgrade) your npm installation?

NdibeRaymond reviewed (commented):
Another thing I wasn't able to comment on directly because the changes are way to much for github to render is the fact that you have so many dirty diffs in the package-lock.json file. This is probably related to the stuff about the issue with your npm version and also maybe related to windows vs linux differences. We should work on fixing this before it can be merged.

Morai-C committed:
changed react scripts back to version 4.0.3 and react quill back to version 1.3.5 and added utils

Morai-C commented:
@tuxology @NdibeRaymond  i just added another commit to address the issues and it works fine now without any errors.. in the new commit i changed the react-scripts and react_quill  back to the default version and also added the utils folder which my npm modified when i ran the npm install command.. everything works fine with npm now and no errors. please review thank you

kamthamc commented on zubhub_frontend/zubhub/package.json:
```diff
@@ -26,14 +26,15 @@
     "redux-thunk": "^2.3.0",
     "slick-carousel": "^1.8.1",
     "source-map-explorer": "^2.5.2",
+    "util": "^0.12.4",
```
Please don’t add libraries. Please revert the changes to package.json & package-lock.json

Morai-C replied:
when i first ran npm install it did some adjustments and the util was part of that.. so it started causing errors because the util was in use but the dependency was no longer there... so i reverted back to the default packages but it still didn't get the util but the errors reduced to just the util so i added it and all the errors cleared cos it could now see the module. anyone running frontend on npm without that would be getting lots of errors and reload issues if their npm messes with that..

kamthamc replied:
I tried to install without docker and I see the package-lock.json was created with old npm and there are some changes in the file. We can create separate issue for updating NodeJS version (in docker) and prerequisites and also update package-lock.json

kamthamc replied:
@tuxology

tuxology replied:
Yes, this is a good idea. I'll let @NdibeRaymond Create this one and describe what to exactly do

kamthamc commented on zubhub_frontend/zubhub/src/assets/js/styles/components/project/projectStyles.js:
```diff
@@ -118,8 +124,9 @@ const styles = theme => ({
     marginBottom: '0.5em',
     display: 'flex',
-    justifyContent: 'flex-start',
+    justifyContent: 'space-between',
```
Please check this in different resolutions. It might look weird if there is too much spacing

Morai-C replied:
yeah did that before adjusting it and after adjusting..

Morai-C committed:
fixed issue of username causing overflow on longer email addresses and also fixes responsiveness on profile page

Morai-C commented:
Description of commit
Closes #367 

Changes
Addresses problem of longer email addresses causing overflow on the profile page, also adjusted reponsiveness of grid items clustering on specific screen resolutions, adjusted comment placeholder text size on smaller screens.
![Screenshot (152)](https://user-images.githubusercontent.com/77807331/162340498-1eccdfff-0ca0-4e71-8166-b0484fb513b8.png)
![Screenshot (148)](https://user-images.githubusercontent.com/77807331/162340504-e0d7930d-9469-4c77-840d-43fbc4520844.png)
![Screenshot (149)](https://user-images.githubusercontent.com/77807331/162340507-f968b438-258f-45c0-81f7-f56e3f28659e.png)
![Screenshot (150)](https://user-images.githubusercontent.com/77807331/162340511-ac712b5c-e333-4653-8a57-eb2075fbc01c.png)
![Screenshot (151)](https://user-images.githubusercontent.com/77807331/162340516-a19234ad-7905-4a14-87d4-457b78e99112.png)

@tuxology  i added a new commit addressing @AndrewLester findings and also made minor adjustments on the responsiveness please review.

Morai-C committed:
fixed overflow on ProjectDetails caused by buttons

tuxology commented:
This looks good and fixes seem to work @Morai-C I am still unsure what do we do with the `package-lock.json` issue for this PR. I suggest you checkout the `package-lock.json` file from master and commit so we can proceed to merge this. Is that the correct workaround @NdibeRaymond?

Morai-C commented:
> This looks good and fixes seem to work @Morai-C I am still unsure what do we do with the `package-lock.json` issue for this PR. I suggest you checkout the `package-lock.json` file from master and commit so we can proceed to merge this. Is that the correct workaround @NdibeRaymond?

please with the package-lock issue i could delete my package-lock and get back another from the master branch and commit or i could run npm install --package-lock-only to resolve conflicts if there are any... which do you suggest i go for? thank you @NdibeRaymond @tuxology

NdibeRaymond commented:
> > This looks good and fixes seem to work @Morai-C I am still unsure what do we do with the `package-lock.json` issue for this PR. I suggest you checkout the `package-lock.json` file from master and commit so we can proceed to merge this. Is that the correct workaround @NdibeRaymond?
> 
> please with the package-lock issue i could delete my package-lock and get back another from the master branch and commit or i could run npm install --package-lock-only to resolve conflicts if there are any... which do you suggest i go for? thank you @NdibeRaymond @tuxology

delete the package-lock in your branch, run `git checkout master -- package-lock.json` to pull the original package-lock from master then commit and push

Morai-C commented:
> > > This looks good and fixes seem to work @Morai-C I am still unsure what do we do with the `package-lock.json` issue for this PR. I suggest you checkout the `package-lock.json` file from master and commit so we can proceed to merge this. Is that the correct workaround @NdibeRaymond?
> > 
> > 
> > please with the package-lock issue i could delete my package-lock and get back another from the master branch and commit or i could run npm install --package-lock-only to resolve conflicts if there are any... which do you suggest i go for? thank you @NdibeRaymond @tuxology
> 
> delete the package-lock in your branch, run `git checkout master -- package-lock.json` to pull the original package-lock from master then commit and push

@tuxology @NdibeRaymond  okay thanks. did that and it says everything is up to date. could you please review and call my attention if there's any issue again. Thank you

NdibeRaymond committed:
Delete package-lock.json

NdibeRaymond committed:
sync package-lock and package.json with master

NdibeRaymond commented:
@Morai-C I updated package-lock.json and package.json myself and tested your PR but there are still certain improvements that needs to be made here. The project tag doesn't appear to work well under certain conditions, the images below:
![creator_tag_issue](https://user-images.githubusercontent.com/40905613/163684876-76639289-008b-4bf2-95f6-29bf15cb793f.PNG)
![creator_tag_issue_2](https://user-images.githubusercontent.com/40905613/163684882-1a7d9c58-c772-49b7-b50a-dad5f99a61b9.PNG)

It'd be better if the tag could wrap under the username, instead of changing the layout of the project card. Also, you can see the `justify center` stuff doesn't look that great.

NdibeRaymond commented:
are you still interested in working on this issue @Morai-C ?

Morai-C commented:
> are you still interested in working on this issue @Morai-C ?

sure thanks @NdibeRaymond i'll get on this right away

Morai-C commented:
> @Morai-C I updated package-lock.json and package.json myself and tested your PR but there are still certain improvements that needs to be made here. The project tag doesn't appear to work well under certain conditions, the images below: ![creator_tag_issue](https://user-images.githubusercontent.com/40905613/163684876-76639289-008b-4bf2-95f6-29bf15cb793f.PNG) ![creator_tag_issue_2](https://user-images.githubusercontent.com/40905613/163684882-1a7d9c58-c772-49b7-b50a-dad5f99a61b9.PNG)
> 
> It'd be better if the tag could wrap under the username, instead of changing the layout of the project card. Also, you can see the `justify center` stuff doesn't look that great.

okay i'll do exactly that @NdibeRaymond

Morai-C committed:
make tag wrap below username  whenever username is long

Morai-C commented:
@NdibeRaymond @tuxology  i just committed again with the latest changes, made the tag wrap due to lengthy usernames, removed the justify content and restored the layout back to its default.. i took off the margins and added grid spacing instead due to the fact that with 3 columns in the grid and mui breakpoints having different values from the original breakpoints when testing the responsiveness the cards touch each and caused minor overlaps.. please review thank you.

![Screenshot (181)](https://user-images.githubusercontent.com/77807331/163725910-b408980f-65a1-4148-b061-895b27088f0d.png)
![Screenshot (176)](https://user-images.githubusercontent.com/77807331/163725912-b5586d64-9606-4654-abc0-8da1c1311181.png)
![Screenshot (177)](https://user-images.githubusercontent.com/77807331/163725915-8fe4d113-2c04-4111-9959-be7048febb41.png)
![Screenshot (178)](https://user-images.githubusercontent.com/77807331/163725916-dcae0e6c-e631-4417-89ec-ccab50230099.png)
![Screenshot (180)](https://user-images.githubusercontent.com/77807331/163725918-3c0cd222-8998-4285-8205-4b47ec646f1b.png)

tuxology commented:
This looks ok to me now. I think its time to merge this @NdibeRaymond ?

kamthamc commented on zubhub_frontend/zubhub/src/assets/js/styles/components/project/projectStyles.js:
```diff
@@ -118,13 +120,23 @@ const styles = theme => ({
+  creatorEmailAndTagBoxStyle: {
+    display: 'flex',
+    flexWrap: 'wrap',
+    [theme.breakpoints.down('376')]: {
```
Its better to use predefined breakpoints like sm or md 

https://mui.com/material-ui/customization/breakpoints/

Morai-C replied:
ok @kamthamc @tuxology  i'll do that now and push

kamthamc replied:
If we want to use custom size we could define a different size while creating the theme and use it here

kamthamc reviewed (commented):
I think its better to use predefined breakpoints instead of adding different values at random places. Apart from it LGTM

Morai-C committed:
Changed projectStyles creatorEmailAndTagBoxStyle breakpoint value to a pre-defined breakpoint

Morai-C commented:
@tuxology please i've made the changes

NdibeRaymond commented:
@Morai-C thanks for all the effort you put into this PR. Though I still feel like there are changes that still need to be made. There are certain changes made to the project card that I don't fully understand the rationale behind. The project card width was reduced for no obvious reason. There is padding bottom on the username even when the user tag is not wrapped. I feel like I need to properly take time to look at the css here.

Morai-C commented:
> @Morai-C thanks for all the effort you put into this PR. Though I still feel like there are changes that still need to be made. There are certain changes made to the project card that I don't fully understand the rationale behind. The project card width was reduced for no obvious reason. There is padding bottom on the username even when the user tag is not wrapped. I feel like I need to properly take time to look at the css here.

@NdibeRaymond @tuxology  thanks for giving me the chance to work on this I am happy to help anytime. I feel the padding-bottom or margin-bottom is a result of the p component by default the p tag has a padding-bottom or margin-bottom.  I adjusted the card because given the allocation of predefined breakpoints when adjusting the browser size to check responsiveness on certain widths with the cards taking 3 columns they either overlapped or were joined without any gap while trying to maintain the width of the parent container. so to maintain the spacing I adjusted the card width and also added a gap. should in case you need me to clarify any other thing let me know.

tuxology commented:
@NdibeRaymond The breakpoints and card width change does make sense to me but I would let you decide if we should be ok with this since it might break UI in multiple places. It is not breaking it on my large and small screens at least. If you feel it is ok, we can request @Morai-C to rebase and resolve conflicts so we can merge?

tuxology commented:
Closing this since card design has substantially changed
